### PR TITLE
Ark Refactor

### PR DESCRIPTION
**Pull request by srivarra**
**If you haven't already, please read through our [contributing guidelines](https://ark-analysis.readthedocs.io/en/latest/_rtd/contributing.html) before opening your PR**

**What is the purpose of this PR?**

Addresses part of #901. Fixed the runner running out of space.

**How did you implement your changes**

- Moved `utils.synthetic_spatial_datagen` to `tests/`
- Moved `utils.spatial_analysis_utils` to `src/ark/analysis/`
- Moved `utils.segmentation_utils` to `src/ark/segmentation/`

After each notebook's tests are finished, the data is cleared.

**Remaining issues**

N/A

### CODE REVIEW

File: tests/utils/notebooks_test.py [hunk 1]
```diff
@@ -1,8 +1,10 @@
+from email.mime import base
```
Comment (ngreenwald) [inline tests/utils/notebooks_test.py:1]: What's this for?

File: tests/utils/deepcell_service_utils_test.py [hunk 2]
```diff
@@ -8,7 +8,10 @@
 import pytest
 import tifffile
 from pytest_mock import MockerFixture
-from alpineer import image_utils
+from alpineer import image_utils, test_utils
+from ark.utils import deepcell_service_utils
+from skimage import io
+# import test_utils
```
Comment (ngreenwald) [inline tests/utils/deepcell_service_utils_test.py:14]: stray import

**Review by ngreenwald — CHANGES_REQUESTED** [review]
Just some minor syntax

File: tests/utils/notebooks_test.py (hunk 1 above)
Comment (srivarra) [reply tests/utils/notebooks_test.py:1]: No idea.

**Review by ngreenwald — APPROVED** [review]
Looks good! Testing out the merge queue again

**Pull request merged**